File: openstack-worker/app/routes/volumes.py
from fastapi import APIRouter
from app.services import volume_service

router = APIRouter(prefix="/api/v1")

@router.get("/volumes")
def list_volumes():
    return volume_service.get_all_volumes_service()

@router.get("/volumes/{volume_id}")
def get_volume(volume_id: str):
    return volume_service.get_volume_by_id_service(volume_id)

@router.delete("/volumes/{volume_id}")
def delete_volume(volume_id: str):
    volume_service.delete_volume_by_id_service(volume_id)
    return {"message": "volume supprimé"}

File: apps/frontend-v2/src/services/volumes.ts
import { api } from '../lib/apiClient'
import type { Volume } from '../types'

export async function getVolumes(): Promise<Volume[]> {
  const { data } = await api.get('/volumes')
  return data
}

export async function getVolumeById(id: string): Promise<Volume> {
  const { data } = await api.get(`/volumes/${id}`)
  return data
}

export async function deleteVolume(id: string): Promise<void> {
  await api.delete(`/volumes/${id}`)
}
